### ISSUE TITLE
Accuracy Limitations

### ISSUE THREAD

**Issue by tolgabirdal** [tbennun/cudnn-training#1]
The code can only achieve ~8% accuracy on the MNIST dataset. Why is that? The original LeNET achieves around ~0.21. There is a huge performance gap.

**Comment by tbennun** [comment]
This is due to the fact that the example is basic, and thus uses the rudimentary (but short) Stochastic Gradient Descent (SGD) method. It does not use Nesterov's Accelerated Gradient method for improving convergence, nor does it set specific optimization weights for the different parameters (i.e., each iteration changes the convolution bias parameters differently than the other convolution coefficients).

Using these methods would increase convergence, at the expense of creating a longer and less-understandable code. Some implementations of Nesterov and other convergence improvement techniques are available in Caffe and Torch.
